polars/polars-lazy/polars-plan/src/logical_plan/mod.rs

```diff
@@ -255,6 +255,10 @@ pub enum LogicalPlan {
         input: Box<LogicalPlan>,
         payload: FileSinkOptions,
     },
+    CloudSink {
```

Comment:
Can we put `Filesink` and `CloudSink` in a single `Sink` node? That makes pattern matching a little more clean.

Reply:
I agree that this would be cleaner. 
So far, I only considered modifying `FileSink` itself (adding more file types); probably because I had no full understanding on how it was used yet.

I think you mean something like

```rust
enum Sink {
  FileSink( ...)
  #[config(feature = "cloud")]
  CloudSink(...)
}
```
?

That would indeed be much nicer! 😄 

Maybe we could even move the `Memory` variant from `FileSink.payload.file_type` to a separate variant in `Sink`?

Reply:
Yes, that's what I meant. :)

> Maybe we could even move the Memory 

Good point. :+1:

Reply:
Done!